src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java

```diff
@@ -57,12 +60,13 @@
 
   private static final Logger log = LoggerFactory.getLogger(HttpClientRequestImpl.class);
 
-  private final String host;
-  private final int port;
+  private String host;
+  private int port;
+  private int redirectCount;
   private final HttpClientImpl client;
-  private final HttpRequest request;
+  private HttpRequest request;
   private final VertxInternal vertx;
-  private final io.vertx.core.http.HttpMethod method;
+  private io.vertx.core.http.HttpMethod method;
```

Comment:
Ok, but then all access to the field must be guarded. It's not the case. So there are race conditions.